### PR TITLE
server: Exit gracefully if no endpoint is local to it.

### PR DESCRIPTION
**Pull request by krisis**
minio server if run with incorrect endpoint arguments as explained in #3438 shoul exit gracefully
## Description

This change checks if any of the supplied endpoint is local to the server. If not it exits gracefully.
## Motivation and Context

See #3438 
## How Has This Been Tested?

Added unit test  and tested using docker-compose setup.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: cmd/server-main.go [hunk 1]
```diff
@@ -361,6 +361,17 @@ func checkServerSyntax(c *cli.Context) {
 	}
 }
 
+func isAnyEndpointLocal(eps []*url.URL) bool {
```
Comment (harshavardhana) [inline cmd/server-main.go:365]: Add a comment here..

File: cmd/server-main.go [hunk 2]
```diff
@@ -403,14 +414,7 @@ func serverMain(c *cli.Context) {
...
-	anyLocalEp := false
-	for _, ep := range endpoints {
-		if isLocalStorage(ep) {
-			anyLocalEp = true
-			break
-		}
-	}
-	if !anyLocalEp {
+	if !isAnyEndpointLocal(endpoints) {
 		fatalIf(errors.New("No endpoint is local to this server"), "No endpoint is local to this server.")
```
Comment (harshavardhana) [inline cmd/server-main.go]: Same error message.. repeated.  Make them distinct.  One is an actual error and another is a user friendly message.

File: cmd/server-main_test.go [hunk 3]
```diff
@@ -347,3 +347,31 @@ func TestInitServerConfig(t *testing.T) {
 		initServerConfig(ctx)
 	}
 }
+
```
Comment (harshavardhana) [inline cmd/server-main_test.go:350]: Comment..

**Review by harshavardhana — CHANGES_REQUESTED** [review]
Added comments.

File: cmd/server-main_test.go (hunk 3 above)
Comment (krisis) [reply cmd/server-main_test.go:350]: done.

File: cmd/server-main.go (hunk 1 above)
Comment (krisis) [reply cmd/server-main.go:365]: done

File: cmd/server-main.go (hunk 2 above)
Comment (krisis) [reply cmd/server-main.go]: done

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**